code/common/merge.q

```diff
@@ -40,8 +41,12 @@ getpartchunks:{[partdirs;mergelimit]
   t:select from .merge.partsizes where ptdir in partdirs;
   /-get list of limits (rowcount or bytesize) to be used to get chunks of partitions to get merged in batch
   r:$[.merge.mergebybytelimit;exec bytes from t;exec rowcount from t];
-  /-return list of partitions to be called in batch by merge by part function
-  (where r={$[z<x+y;y;x+y]}\[0;r;mergelimit]) cut exec ptdir from t
+  /-return list of partitions to be called in batch
+  l:(where r={$[z<x+y;y;x+y]}\[0;r;mergelimit]),(select count i from .merge.partsizes)[`x];
+  /-where there are more than set partlimit, split the list
+  s:-1_ asc raze {$[(x[y]-x[y-1])<=.merge.partlimit;x[y];x[y], x[y-1] + (v where 0=(v:1+ til x[y] - x[y-1]) mod .merge.partlimit)]}/:[l;til count l];
```

Comment:
I think this can be simplified a bit e.g. something like

```suggestion
  s:-1_ raze {$[(x[y]-x[y-1])<=.merge.partlimit;x[y];.merge.partlimit cut x[y-1]+til x[y] - x[y-1]]}/:[l;til count l];
```

don't think the `asc` is needed with this version as it'll still be in order anyway

Reply:
Yes the new version doesn't need sorted. The output isn't exactly the same format, it gives a matrix rather than list, so I will make changes to it but it should still be a simplified version of the initial commit

Reply:
```suggestion
  s:-1_distinct asc raze {$[(x[y]-x[y-1])<=.merge.partlimit;x[y];x[y], raze 1#'.merge.partlimit cut x[y-1]+til x[y] - x[y-1]]}/:[l;til count l];
```
This is a simplified version based on the above suggestion. After the cut and indexing into the matrix I still needed to add the last value of l, which then needs to be sorted with asc and in this case remove duplicates. The extra steps are to stop any of the chunks made being an empty list.

Reply:
instead of `raze 1#'` I think `first each` would do the same and probably clearer

```suggestion
  s:-1_distinct asc raze {$[(x[y]-x[y-1])<=.merge.partlimit;x[y];x[y], first each .merge.partlimit cut x[y-1]+til x[y] - x[y-1]]}/:[l;til count l];
```

Reply:
Yes that works the same, agree that is a clearer way to do it